File: java/java-arithmetic/src/main/java/com/leetcode/first/J53.java
package com.leetcode.first;

public class J53 {

    public static void main(String[] args) {
        System.out.println(new J53().maxSubArray(new int[]{-2,2,1,-3,4,-1,2,1,-5,4}));

    }

    public int maxSubArray(int[] nums) {
        if (nums==null||nums.length==0){
            return 0;
        }
        int max =nums[0];
        for (int i=0;i<nums.length;i++){
            int num=0;
            for (int j=i;j<nums.length;j++){
                 num = num+nums[j];
                 max = Math.max(max,num);
            }
        }
        return max;
    }
}

File: python-test/leetcode/J53.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        maxNum=nums[0]
        for i in range(len(nums)):
            num = 0
            for j in range(i,len(nums)):
                num = num+ nums[j]
                maxNum =max(num,maxNum)
        
        return maxNum
    
    def maxSubArray2(self, nums: list[int]) -> int:
        prex=0
        maxAns=nums[0]
        for i in nums:
            prex =max(prex+i,i)
            maxAns=max(maxAns,prex)
        
        return maxAns

